File: Examples/VisualBasic/Working-Diagrams/RetrieveFontInfo.vb
Imports Microsoft.VisualBasic
Imports System.IO
Imports Aspose.Diagram
Imports System

Namespace Diagrams
    Public Class RetrieveFontInfo
        Public Shared Sub Run()
            ' ExStart:RetrieveFontInfo
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Diagrams()

            ' Call the diagram constructor to load diagram from a VSD file
            Dim vdxDiagram As New Diagram(dataDir & "RetrieveFontInfo.vsd")

            For Each font As Aspose.Diagram.Font In vdxDiagram.Fonts
                ' Display information about the fonts
                Console.WriteLine(font.Name)
            Next font
            ' ExEnd:RetrieveFontInfo
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Working-Diagrams/RetrieveFontInfo.cs
using System.IO;
using Aspose.Diagram;
using System;

namespace Aspose.Diagram.Examples.CSharp.Diagrams
{
    public class RetrieveFontInfo
    {
        public static void Run()
        {
            // ExStart:RetrieveFontInfo
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Diagrams();

            // Call the diagram constructor to load diagram from a VSD file
            Diagram vdxDiagram = new Diagram(dataDir + "RetrieveFontInfo.vsd");

            foreach (Aspose.Diagram.Font font in vdxDiagram.Fonts)
            {
                // Display information about the fonts
                Console.WriteLine(font.Name);
            }
            // ExEnd:RetrieveFontInfo
        }
    }
}